src/useTimer.ts

```diff
@@ -16,43 +17,38 @@ export interface IValues {
   time: number;
 }
 
-const initialConfig = {
-  endTime: null,
+const initialConfig: IConfig = {
+  endTime: 0,
   initialTime: 0,
   interval: 1000,
   timerType: 'INCREMENTAL',
+  step: 1
 };
 
-export const useTimer = (config?: IConfig): IValues => {
-  const { endTime, initialTime, interval, timerType } = {
+export const useTimer = (config: Partial<IConfig> = initialConfig): IValues => {
```

Comment:
OK for Partial.

For defaulting, i don't think it's a good idea because if no config is provided, config will be defaulted to initialConfig so merge will be:
{
  ...initialConfig,
  ...config, // Same value as initialConfig, with defaulting! These two objects are identical.
}